internal/services/containers/container_registry_cache_rule_resource.go

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package containers
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/containerregistry/2021-08-01-preview/registries"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/containerregistry/2023-07-01/cacherules"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+)
+
+var _ sdk.Resource = ContainerRegistryCacheRule{}
+
+type ContainerRegistryCacheRule struct{}
+
+func (ContainerRegistryCacheRule) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			Description:  "The name of the cache rule.",
+			ValidateFunc: validate.ContainerRegistryCacheRuleName,
+		},
+		"container_registry_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: registries.ValidateRegistryID,
+		},
+		"credential_set_id": {
```

Comment:
What is this exactly, I wasn't able to find any reference to it in the tests? Can this be a key vault store?

Reply:
Update: Just read through the PR description, please disregard.

However just because we don't have this as an available resource in the provider doesn't mean that we can use the validation functions from 
"github.com/hashicorp/go-azure-sdk/resource-manager/containerregistry/2023-07-01/credentialsets"